include/odp/api/spec/traffic_mngr.h

```diff
@@ -246,6 +246,29 @@ typedef struct {
 	odp_bool_t weights_supported;
 } odp_tm_level_capabilities_t;
 
+/** The tm_pkt_prio_mode_t enumeration type is used to indicate different
+ * modes a tm system supports with respect to assigning priority to a packet
+ * and propagating it across TM system. All the nodes in a TM system can
+ * function only on single mode specified at time of odp_tm_create().
+ */
+typedef enum odp_tm_pkt_prio_mode {
+	/** Indicates Packet priority global mode. In this mode, a packet gets its priority
+	 * based on a TM queue it gets enqueued to and then it carries the same priority along
+	 * with it as long as it is in the TM system. At every TM node in the topology,
+	 * that specific pkt is scheduled as per that priority.
+	 */
+	ODP_TM_PKT_PRIO_MODE_GLOBAL,
+	/** Indicates Packet priority override mode. In this mode, a packet gets a new priority
+	 * every time it passes through a TM queue or a TM node. All the packets fed by a fan-in
+	 * node will get the same priority and that will be valid until it reaches the next
+	 * TM node. This priority is part of the TM fan-in node parameters and is fixed
```

Comment:
> will be valid until it reaches the next TM node.

Here "next TM node" does not mean the next TM node after the one which overrides the priority (since that is the one that looks at the priority) but the one after that. Maybe this should be: "... valid until overridden again by another TM node".

Btw, the term override does not sound quite right to me but I am not sure what I would propose instead.

Reply:
How about ODP_TM_PKT_PRIO_MODE_PER_LVL  ?

Reply:
I think _PER_LVL is confusing. Also, the _GLOBAL does not sound quite right to me.

How about _PRIO_PRESERVE and_PRIO_OVERWRITE? Or _PRIO_FIXED and _PRIO_NODE?

Reply:
PRESERVE and OVERWRITE sounds fine to me. I'll make the change in v3.